### PR TITLE
Add pytorch implementation of gan workshop

### PR DESCRIPTION
**Pull request by abster95**
-Move the old code to legacy folder
-Add pytorch implementation
-This request contains more code than it needs. Will delete some stuff for the participants to fill in. Creating PR with full code for reviewer's context

### CODE REVIEW

File: workshops/gan/src/training.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+assert MAX_SUMMARY_IMAGES <= BATCH_SIZE
+
+def train():
+    summary_writer = SummaryWriter()
+
+    critic = FCCritic(IMG_SIZE, CHANNELS)
+    generator = FCGenerator(IMG_SIZE, CHANNELS, Z_SIZE)
+    if CUDA:
+        critic.cuda()
+        generator.cuda()
```
Comment (GoranDubajic) [inline workshops/gan/src/training.py:32]: Maybe to set up at the begining
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
and then
critic.to(device)
generator.to(device)
etc.

File: workshops/gan/src/generators.py [hunk 2]
```diff
@@ -1,86 +1,31 @@
...
 
-            z = tf.layers.dense(z, 32768, activation=act)
-            z = tf.reshape(z, [-1, 4, 4, 2048])
+    def forward(self, z_batch):
+        x = self.fc1(z_batch)
+        x = self.relu(x)
+        x = self.fc2(x)
+        x = self.relu(x)
+        x = self.fc3(x)
+        x = self.relu(x)
```
Comment (GoranDubajic) [inline workshops/gan/src/generators.py:27]: The last activation should be sigmoid so we get output in range [0, 1]?

Comment (abster95) [reply workshops/gan/src/generators.py:27]: This is a generator, I don't think it matters all that much. I'll add it just in case
